ipa-core/src/ff/boolean_array.rs

```diff
@@ -138,7 +138,7 @@ macro_rules! boolean_array_impl {
                     let v = v.into();
                     let mut val = Self::ZERO;
                     for i in 0..$bits {
```

Comment:
This is better, though I note that `min` is not `const`.  I expect that the compiler is capable of dealing with that though.

```suggestion
                    for i in 0..std::cmp::min(128,$bits) {
```

Note that this is faster because it is a single comparison, not `$bits` comparisons.  I think that it also makes intent clearer, to the point Alex makes below.